File: Cpp/1800_Maximum_Ascending_Subarray_Sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int sum = nums[0];
        int result = sum;
        
        for(int i=1; i<nums.size(); ++i) {
            if(nums[i] <= nums[i-1]) {
                sum = 0;
            }
            sum += nums[i];
            result = max(result, sum);
        }
        
        return result;
    }
};

File: Kotlin/1800_Maximum_Ascending_Subarray_Sum.kt
class Solution {
    fun maxAscendingSum(nums: IntArray): Int {
        var sum = nums[0]
        var result = sum
        
        for(i in 1 until nums.size) {
            if(nums[i] <= nums[i - 1]) {
                sum = 0
            }
            sum += nums[i]
            result = maxOf(result, sum)
        }
        
        return result
    }
}